### PR TITLE
At ftre

### PR DESCRIPTION
**Pull request by TechnichalDoge**
Im confused about the table version of this, and unfortunately for me, i cannot gleam information from this https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/table.lua#L733, so i will have to find out how to treat the array part like an array.

### CODE REVIEW

**Comment by TechnichalDoge** [comment]
By the by, this is for #2

**Comment by TechnichalDoge** [comment]
I completed the table portion, but i think its not bug free, but i dont see any bugs, so please check it. Thanks

File: lua/entities/gmod_wire_expression2/core/custom/cfc_array.lua [hunk 1]
```diff
@@ -90,3 +90,20 @@ e2function number array:indexOf(vector4 vec4)
     return indexOfPackedValue( this, vec4 )
 end
 
+e2function number array:indexOf(vector4 vec4)
+    return indexOfPackedValue( this, vec4 )
+end
+
+registerFunction( "first", "r:", id, function(self,args)
+  local op1 = args[2]
```
Comment (PLally) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_array.lua]: 4 space indents

Comment (PLally) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_array.lua]: could you use more descriptive variable names

Comment (TechnichalDoge) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_array.lua]: i did just use the shift function as a template, cut me some slack, anyway will do

**Comment by TechnichalDoge** [comment]
I fixed the spaces and the variable names are better

File: lua/entities/gmod_wire_expression2/core/custom/cfc_array.lua (hunk 1 above)
Comment (TechnichalDoge) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_array.lua]: Already done.

**Comment by TechnichalDoge** [comment]
This needs a rework

**Pull request closed without merging**